librosa/core/audio.py

```diff
@@ -469,6 +469,114 @@ def autocorrelate(y, max_size=None, axis=-1):
     return autocorr
 
 
+@jit
+def lpc(y, m):
+    """Linear Predictive Coefficients via Burg's method
+
+    This function applies Burg's method to estimate coefficients of a linear
+    filter on `y` of order `m`.  Burg's method is an extension to the
+    classic Yule-Walker approach which is sometimes referred to as
+    the LPC `autocorrelation` method.
+
+    It follows the description and implementation approach described in the
+    introduction in [1].  (Note that this paper describes a different method,
+    which is not implemented here, but has been chosen for its clear
+    explanation of Burg's technique in its introduction.)
+
+    .. [1] Larry Marple
+           A New Autoregressive Spectrum Analysis Algorithm
+           IEEE Transactions on Accoustics, Speech and Signal Processing
+           Vol ASSP-28, No. 4, August 1980
+
+    Parameters
+    ----------
+    y : np.ndarray
+        Time series to fit
+
+    m : int > 0
+        Order of the linear filter
+
+    Returns
+    -------
+    a : np.ndarray of length m + 1
+        LPC filter coefficients (ie, denominator polynomial of the filter)
+    k : np.ndarray of length m
+        Reflection coefficients
+
+    Raises
+    ------
+    ParameterError
+        - If y is not real valued
+        - If m < 1 or not integer
+
+    See also
+    --------
+    scipy.signal.lfilter
+
+    Examples
+    --------
+    Compute LPC coefficients of y at order 16 on entire series
+    >>> y, sr = librosa.load(librosa.util.example_audio_file(), offset=20,
+    ...                      duration=10)
+    >>> librosa.lpc(y, 16)
+
+    Compute LPC coefficients, and plot LPC estimate of original series
+    >>> import matplotlib.pyplot as plt
+    >>> y, sr = librosa.load(librosa.util.example_audio_file(), offset=20,
+    ...                      duration=0.020)
+    >>> a, _ = librosa.lpc(y, 16)
+    >>> y_est = scipy.signal.lfilter([1], a, y)
+    >>> plt.plot(y)
+    >>> plt.plot(y_est)
+
+    """
+    if not isinstance(m, int) or m < 1:
+        raise ParameterError("m must be an integer > 0")
+
+    if not isinstance(y, np.ndarray):
+        raise ParameterError("y must be of the type np.ndarray")
+
+    if not all(np.isreal(y)):
+        raise ParameterError("y must not contain complex numbers")
+
+    fp = bp = y
+    k = np.zeros(m)
+    a = np.zeros(m+1)
+    den = 0
+    q = a[0] = 1
+
+    for i in range(m):
+        # Remove first and last element from forward and reverse
+        # prediction errors
+        fp = fp[1:]
+        bp = bp[:-1]
+
+        # Compute autocorrelation (denominator), use last result
+        # to save on computation if we have one
+        #
+        # This does cause a slight difference in the result,
+        # likely from numerical issues, but in practice it's
+        # neglible
+        if i == 0:
+            den = np.dot(fp, fp) + np.dot(bp, bp)
```

Comment:
i would prefer to have the zeroth iteration completely outside of the loop rather than checking for `i==0` at every iteration

Reply:
Fixed in 7bab19f